### PR TITLE
dnspeep: init at 0.1.2

### PR DESCRIPTION
**Pull request by figsoda**
###### Motivation for this change

`dnspeep` lets you spy on the DNS queries your computer is making.

https://github.com/jvns/dnspeep

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/dnspeep/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    sha256 = "0lz22vlgi1alsq676q4nlzfzwnsrvziyqdnmdbn00rwqsvlb81q6";
+  };
+
+  cargoLock = {
+    # update Cargo.lock every update
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "pcap-0.8.1" = "1nnfyhlhcpbl4i6zmfa9rvnvr6ibg5khran1f5yhmr9yfhmhgakd";
+    };
+  };
```
Comment (SuperSandro2000) [inline pkgs/tools/security/dnspeep/default.nix:20]: Why can't we use the upstream one?

Comment (figsoda) [reply pkgs/tools/security/dnspeep/default.nix:20]: using `fetchurl` as `cargoLock.lockFile` utilizes ifd and upsets ofborg

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133157` at fc71071f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>dnspeep</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/tools/security/dnspeep/default.nix:1:48:
```
  |
1 | { fetchFromGitHub, lib, libpcap, rustPlatform, stdenv }:
  |                                                ^
```
</li>
  </ul>
</details>

File: pkgs/tools/security/dnspeep/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/security/dnspeep/default.nix:20]: The upstream repo has a lock file included.

Comment (figsoda) [reply pkgs/tools/security/dnspeep/default.nix:20]: `cargoLock` is an alternative to `cargoSha256`, not `Cargo.lock`. upsream lock file includes a git dependency, which doesn't work with `cargoSha256`, so I have to download `Cargo.lock` and use `cargoLock` as an alternative

Comment (SuperSandro2000) [reply pkgs/tools/security/dnspeep/default.nix:20]: Can you add a comment that explains that the git dependency requires this? Thank you!

Comment (figsoda) [reply pkgs/tools/security/dnspeep/default.nix:21]: added

**Pull request merged**